libpod/options.go

```diff
@@ -548,14 +553,78 @@ func WithRuntimeFlags(runtimeFlags []string) RuntimeOption {
 
 // Container Creation Options
 
+// WithCpuSetCpusCtr computes and sets the Cpus linux resource string which determines the amount of cores, from those available,  we are allowed to execute on
+func WithCPUSetCPUsCtr(res string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return define.ErrRuntimeFinalized
+		}
+
+		wholeNum := ""
+		for j := 0; j < len(res); j++ {
+			temp := res[j]
+			tempS := string(temp)
+			if len(wholeNum) == 0 || len(wholeNum) == 1 {
+				wholeNum = tempS
+			}
+			if val, err := strconv.Atoi(wholeNum); err == nil {
+				if int(runtime.NumCPU()) < val {
+					max := fmt.Sprint(runtime.NumCPU())
+					totalNew := "0-" + max
+					res = totalNew
+				}
+			}
+			if j != len(res)-1 {
+				if (string(res[j+1]) != "-") && (string(res[j+1]) != ",") && wholeNum != "-" && wholeNum != "," {
+					wholeNum += string(res[j+1])
+				}
+			}
+		}
+		ctr.config.Spec.Linux.Resources.CPU.Cpus = res
+		return nil
+	}
+}
+
+// WithCpusCtr computes the CPU period and quota given the CPU core count for a container
+func WithCPUsCtr(num float64) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return define.ErrRuntimeFinalized
+		}
+		if ctr.config.Pod != "" {
+			podID := ctr.config.Pod
+			pod, err := ctr.runtime.state.Pod(podID)
+			if err != nil {
+				return errors.New("could not get cpu data")
```

Comment:
```suggestion
				return errors.Wrap(err, "could not get cpu data")
```
Using https://pkg.go.dev/github.com/pkg/errors#Wrap or Wrapf will pass error information to caller vs throwing it away.